src/Event/ApplyResultsEvent.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Thunder\Shortcode\Event;
+
+use Thunder\Shortcode\Shortcode\ShortcodeInterface;
+
+/**
+ * @author Tomasz Kowalczyk <tomasz@kowalczyk.cc>
+ */
+class ApplyResultsEvent
+{
+    private $shortcode;
+    private $text;
+    private $replaces;
+    private $result;
+
+    public function __construct(ShortcodeInterface $shortcode = null, $text, array $replaces)
```

Comment:
Make those replaces more explicit, right now they are internal raw array directly from the heart of `Processor`. Maybe `addReplacement()` or something?